File: FrontEnd/ReactUser/src/models/doctor.ts
export class Doctor {
  DoctorId?: number
  DoctorName = ""
  FieldOfSpecialization = ""
  LicenseNumber = 0


  constructor(data: any = {}) {
    this.DoctorId = data.DoctorId !== undefined ? Number(data.DoctorId) : undefined
    this.DoctorName = data.DoctorName || ""
    this.FieldOfSpecialization = data.FieldOfSpecialization || ""
    this.LicenseNumber = data.LicenseNumber || 0
  }
}


export class DoctorPost {
  DoctorName = ""
  FieldOfSpecialization = ""
  LicenseNumber = 0

  constructor(data: any = {}) {
    this.DoctorName = data.DoctorName || ""
    this.FieldOfSpecialization = data.FieldOfSpecialization || ""
    this.LicenseNumber = data.LicenseNumber || 0
  }
}

File: BackEnd/FinalProject/FinalProject.Clean/Models/Doctor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FinalProject.Core.Models
{
    public class Doctor
    {
        [Key]
        public int DoctorId { get; set; }
        [Required]
        [MaxLength(100)]
        public string DoctorName { get; set; }
        [Required]
        [MaxLength(100)]
        public string FieldOfSpecialization { get; set; }
        [Required]
        public int LicenseNumber { get; set; }

        public Doctor(int doctorId, string doctorName, string fieldOfSpecialization, int licenseNumber)
        {
            DoctorId = doctorId;
            DoctorName = doctorName;
            FieldOfSpecialization = fieldOfSpecialization;
            LicenseNumber = licenseNumber;
        }

        public Doctor (string doctorName, string fieldOfSpecialization, int licenseNumber)
        {
            DoctorName = doctorName;
            FieldOfSpecialization = fieldOfSpecialization;
            LicenseNumber = licenseNumber;
        }


    }
}
